mod-source-record-manager-server/src/main/java/org/folio/dao/JobExecutionProgressDaoImpl.java

```diff
@@ -3,146 +3,101 @@
 import io.vertx.core.AsyncResult;
 import io.vertx.core.Future;
 import io.vertx.core.Promise;
+import io.vertx.sqlclient.RowIterator;
+import io.vertx.sqlclient.Tuple;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import io.vertx.sqlclient.Row;
 import io.vertx.sqlclient.RowSet;
 
 import java.util.Optional;
-import java.util.UUID;
 import java.util.function.UnaryOperator;
 import javax.ws.rs.NotFoundException;
 
-import org.folio.cql2pgjson.exception.FieldException;
 import org.folio.dao.util.PostgresClientFactory;
 import org.folio.rest.jaxrs.model.JobExecutionProgress;
-import org.folio.rest.persist.Criteria.Criteria;
-import org.folio.rest.persist.Criteria.Criterion;
 import org.folio.rest.persist.PostgresClient;
 import org.folio.rest.persist.SQLConnection;
-import org.folio.rest.persist.cql.CQLWrapper;
-import org.folio.rest.persist.interfaces.Results;
 import org.folio.rest.tools.utils.ValidationHelper;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Repository;
 
-import static org.folio.dataimport.util.DaoUtil.constructCriteria;
-import static org.folio.dataimport.util.DaoUtil.getCQLWrapper;
+import static org.folio.rest.persist.PostgresClient.convertToPsqlStandard;
 
 @Repository
 public class JobExecutionProgressDaoImpl implements JobExecutionProgressDao {
 
   private static final Logger LOGGER = LogManager.getLogger();
 
+  public static final String JOB_EXECUTION_ID_FIELD = "job_execution_id";
+  public static final String TOTAL_RECORDS_COUNT = "total_records_count";
+  public static final String PROCESSED_RECORDS_COUNT = "processed_records_count";
+  public static final String ERROR_RECORDS_COUNT = "error_records_count";
+
   private static final String TABLE_NAME = "job_execution_progress";
-  public static final String JOB_EXECUTION_ID_FIELD = "jobExecutionId";
+  private static final String INSERT_SQL = "INSERT INTO %s.%s (job_execution_id, total_records_count, processed_records_count, error_records_count) VALUES ($1, $2, $3, $4)";
+  private static final String UPDATE_SQL = "UPDATE %s.%s SET job_execution_id = $1, total_records_count = $2, processed_records_count = $3, error_records_count = $4 WHERE job_execution_id = $1";
+  private static final String SELECT_BY_JOB_EXECUTION_ID_QUERY = "SELECT * FROM %s.%s WHERE job_execution_id = $1";
+  private static final String SELECT_FROM_JOB_EXECUTION_PROGRESS = "SELECT * FROM %s.%s WHERE job_execution_id = $1 LIMIT 1 FOR UPDATE";
```

Comment:
Can we combine these 2 queries into single select query
SELECT_BY_JOB_EXECUTION_ID_QUERY 
SELECT_BY_JOB_EXECUTION_ID_QUERY ?

Reply:
Thank you @SerhiiNosko , done. Also removed unnecessary compose call.